From 63ba6280172da7cfdfe5f4f1481dde0e3c8bed9d Mon Sep 17 00:00:00 2001
From: Michael Stack <stack@apache.org>
Date: Fri, 6 Apr 2018 14:25:07 -0700
Subject: [PATCH] HBASE-20149 Purge dev javadoc from bin tarball (or make a
 separate tarball of javadoc)

Instead point back to the website where version-specific dev apis are
hosted.
---
 hbase-assembly/src/main/assembly/components.xml | 5 +++++
 src/main/site/site.xml                          | 4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/hbase-assembly/src/main/assembly/components.xml b/hbase-assembly/src/main/assembly/components.xml
index fe570780ae..2eb16e7649 100644
--- a/hbase-assembly/src/main/assembly/components.xml
+++ b/hbase-assembly/src/main/assembly/components.xml
@@ -26,6 +26,11 @@
     <fileSet>
       <directory>${project.basedir}/../target/site</directory>
       <outputDirectory>docs</outputDirectory>
+      <!--Don't include dev apidocs. They bloat the tgz by almost 50%.
+           See HBASE-20149-->
+      <excludes>
+        <exclude>**/*devapidocs/**</exclude>
+      </excludes>
     </fileSet>
     <!-- Include top level text files-->
     <fileSet>
diff --git a/src/main/site/site.xml b/src/main/site/site.xml
index 400ae8c5d7..f0367027a2 100644
--- a/src/main/site/site.xml
+++ b/src/main/site/site.xml
@@ -100,8 +100,8 @@
       <item name="Getting Started" href="book.html#quickstart" target="_blank" />
       <item name="User API" href="apidocs/index.html" target="_blank" />
       <item name="User API (Test)" href="testapidocs/index.html" target="_blank" />
-      <item name="Developer API" href="devapidocs/index.html" target="_blank" />
-      <item name="Developer API (Test)" href="testdevapidocs/index.html" target="_blank" />
+      <item name="Developer API" href="https://hbase.apache.org/2.0/devapidocs/index.html" target="_blank" />
+      <item name="Developer API (Test)" href="https://hbase.apache.org/2.0/testdevapidocs/index.html" target="_blank" />
       <item name="中文参考指南(单页)" href="http://abloz.com/hbase/book.html" target="_blank" />
       <item name="FAQ" href="book.html#faq" target="_blank" />
       <item name="Videos/Presentations" href="book.html#other.info" target="_blank" />
-- 
2.16.3
